projects/packages/waf/src/class-waf-standalone-bootstrap.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Handles generation and deletion of the bootstrap for the standalone WAF mode.
+ *
+ * @package automattic/jetpack-waf
+ */
+
+namespace Automattic\Jetpack\Waf;
+
+use Exception;
+
+/**
+ * Handles the bootstrap.
+ */
+class WafStandaloneBootstrap {
+
+	/**
+	 * Ensures that constants are initialized if this class is used.
+	 */
+	public function __construct() {
+		$this->guard_against_missing_abspath();
+		$this->initialize_constants();
+	}
+
+	/**
+	 * Ensures that this class is not used unless we are in the right context.
+	 *
+	 * @return void
+	 * @throws Exception If we are outside of WordPress.
+	 */
+	private function guard_against_missing_abspath() {
+
+		if ( ! defined( 'ABSPATH' ) ) {
+			throw new Exception( 'Cannot generate the WAF bootstrap if we are not running in WordPress context.' );
+		}
+	}
+
+	/**
+	 * Does stuff?
+	 *
+	 * @return void
+	 */
+	private function initialize_constants() {
```

Comment:
The thing is that the actions.php is not executed when the WAF runs in standalone mode. See also [my comment about the run.php](https://github.com/Automattic/jetpack/pull/23076#discussion_r817823364).

`JETPACK_WAF_DIR` is actually defined in the actions.php by reading it from the option in case it did not exist before. However, defining `JETPACK_WAF_WPCONFIG` is not really necessary in "plugin mode" (using actions.php) because we are already running inside of WordPress in this case.